app/src/main/java/com/amaze/filemanager/asynchronous/asynctasks/movecopy/PrepareCopyTask.java

```diff
@@ -321,6 +330,29 @@ private void replaceFiles(
     onEndDialog(path, filesToCopy, conflictingFiles);
   }
 
+  private void resolveByRenaming(
+      ArrayList<HybridFileParcelable> filesToCopy,
+      ArrayList<HybridFileParcelable> conflictingFiles) {
+    int appendInt = 1;
+    String newName;
+    File targetFile;
+
+    int conflictingFileIndex = filesToCopy.indexOf(conflictingFiles.get(0));
+    String oldName = filesToCopy.get(conflictingFileIndex).getName();
+    do {
+      newName =
+          oldName.substring(0, oldName.lastIndexOf("."))
+              + "("
```

Comment:
@TranceLove  @VishnuSanal your inputs here. Do we have a library or a utility class that we can utilise for this?

Reply:
Related #2078 sorry but still need some time to work on it.

Reply:
You may want to see if #3913 can help.